src/mainwindow/mainwindow.cpp

```diff
@@ -799,8 +800,10 @@ void MainWindow::saveMainWindowState()
 void MainWindow::restoreMainWindowState()
 {
     QSettings settings(QApplication::organizationName(), QApplication::applicationName());
-    restoreGeometry(settings.value("mainWindowGeometry").toByteArray());
-    restoreState(settings.value("mainWindowState").toByteArray());
+    if (settings.contains("mainWindowGeometry"))
+        restoreGeometry(settings.value("mainWindowGeometry").toByteArray());
+    if (settings.contains("mainWindowState"))
+        restoreState(settings.value("mainWindowState").toByteArray());
```

Comment:
you do the same two times. Just create a function like
void f(string str)
{
if (settings.contains("str"))
        restoreGeometry(settings.value("str").toByteArray());
}